src/App.js

```diff
@@ -1,19 +1,295 @@
+  }, [selectedGenre, sortBy, query, searchString])
+
+  const updateCurrentURL=()=>{
+    const queryParams ={activeGenre: selectedGenre, sortBy: sortBy};
+    if(searchString){
+      queryParams['query'] = searchString;
+    }
+    setSearchParams(queryParams);
+  }
+
+  const callMoviewAPI = (selectedGenre, sortBy,  searchString) =>{
+    let searchQuerParam = new URLSearchParams();
+    searchQuerParam.set("offset", 0);
+    searchQuerParam.set("limit", 20);
+    if(selectedGenre && !searchString){
+      searchQuerParam.set("filter", selectedGenre);
+    }
+    searchQuerParam.set("sortBy", sortBy);
+    searchQuerParam.set("sortOrder", "desc");
+    if (searchString) {
+      searchQuerParam.set("searchBy", "title");
+      searchQuerParam.set("search", searchString);
+    }
+    axios.get(HOST+"/movies?"+searchQuerParam.toString())
+    .then((response) => {
+      setMovies(response.data.data);
+    })
+    .catch((err) => {
+      console.log('error', err)
+     });
+  }
+
+  const createSearchParamsCustom = ()=>{
+    const queryParams ={activeGenre: selectedGenre, sortBy: sortBy};
+    if(searchString){
+      queryParams['query'] = searchString;
+    }
+    return createSearchParams(queryParams);
+  }
+  const onGenreSelected = (selectedGenre) => {
+    console.log("You selected: "+ selectedGenre)
+    setSearchString(null)
+    setSelectedGenre(selectedGenre);
+    navigate({
+      pathname: "/",
+      search: createSearchParamsCustom().toString()
+    });
+  }
+
+  const onTileSelected = (movieData) => {
+    navigate("/movie/"+movieData.id)
+    setSelectedMovieData(movieData)
+    toggleMovieInfoPanel(true)
+  }
+
+  const toggleMovieInfoPanel = (show) =>{
+    window.scrollTo(0, 0)
+    setShowMovieInfoPanel(show)
+    console.log("toggleMovieInfoPanel called: ", show)
+  }
+
+  const viewMovieData = (movieData) => {
+    console.log("viewMovieData: ", movieData)
+    // alert("viewMovieData: "+ movieData)
+    var str="\n\n";
+    for (const [key, value] of Object.entries(movieData)) {
+      str = str + key+ " : "+ value + "\n";
+    }
+    alert("=====viewMovieData======="+ str)
+  }
+
+  const addMovieData = () => {
+    navigate("/movie/add")
+  }
+  
+  const editMovieData = (movieData) => {
+    console.log("editMovieData: ", movieData)
+    // alert("editMovieData: "+ movieData)
+    var str="\n\n";
+    for (const [key, value] of Object.entries(movieData)) {
+      str = str + key+ " : "+ value + "\n";
+    }
+    // alert("=====editMovieData======="+ str)
+    
+    navigate("/"+movieData.id+"/edit?"+fetchCurrentQueryParams(searchParams))
+  }
+
+  const deleteMovieData = (movieData) => {
+    console.log("deleteMovieData: ", movieData)
+    // alert("deleteMovieData: "+ movieData)
+    var str="\n\n";
+    for (const [key, value] of Object.entries(movieData)) {
+      str = str + key+ " : "+ value + "\n";
+    }
+    // alert("=====deleteMovieData======="+ str)
+    openDeleteMovieDialog(movieData)
+  }
+
```

Comment:
Same comments for the repeated or similar code, on using const, string interpolation